crates_io_worker/src/util.rs

```diff
@@ -3,6 +3,22 @@ use sentry_core::Hub;
 use std::any::Any;
 use std::future::Future;
 use std::panic::PanicInfo;
+use tokio::task::JoinError;
+
+pub async fn spawn_blocking<F, R, E>(f: F) -> Result<R, E>
+where
+    F: FnOnce() -> Result<R, E> + Send + 'static,
+    R: Send + 'static,
+    E: Send + From<JoinError> + 'static,
+{
+    let hub = Hub::current();
+    tokio::task::spawn_blocking(move || Hub::run(hub, f))
+        .await
+        // Convert `JoinError` to `E`
+        .map_err(Into::into)
+        // Flatten `Result<Result<_, E>, E>` to `Result<_, E>`
+        .and_then(std::convert::identity)
```

Comment:
I always forget `identity` exists! (My usual pattern there is `Ok(...await??)`, but I don't think that's actually any better. Just interesting seeing the different style.)